src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs

```diff
@@ -21,6 +22,18 @@ static TelemetryCorrelationHttpModule()
             onStepMethodInfo = typeof(HttpApplication).GetMethod("OnExecuteRequestStep");
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether TelemetryCorrelationHttpModule should parse headers by default to get correlation ids.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static bool ParseHeadersDefault { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether TelemetryCorrelationHttpModule should parse headers to get correlation ids.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool? ParseHeaders { get; set; } = null;
```

Comment:
```suggestion
        public bool ParseHeaders { get; set; } = ParseHeadersDefault;
```

then there is no need in `ParseHeaders ?? ParseHeadersDefault`

Reply:
reverted the static variable for initial value.